modelling/src/main/java/org/axonframework/modelling/entity/PolymorphicEntityMessagingMetamodel.java

```diff
@@ -41,72 +41,72 @@
+ * @param <E> The type of polymorphic entity this metamodel represents.
  * @author Mitchell Herrijgers
  * @since 5.0.0
  */
-public class PolymorphicEntityModel<E> implements EntityModel<E>, DescribableComponent {
+public class PolymorphicEntityMessagingMetamodel<E> implements EntityMessagingMetamodel<E>, DescribableComponent {
 
-    private final EntityModel<E> superTypeModel;
-    private final Map<Class<? extends E>, EntityModel<? extends E>> concreteModels;
+    private final EntityMessagingMetamodel<E> superTypeMetamodel;
+    private final Map<Class<? extends E>, EntityMessagingMetamodel<? extends E>> concreteMetamodels;
     private final Set<QualifiedName> supportedCommandNames = new HashSet<>();
     private final Set<QualifiedName> supportedInstanceCommandNames = new HashSet<>();
     private final Set<QualifiedName> supportedCreationalCommandNames = new HashSet<>();
 
-    private PolymorphicEntityModel(
-            EntityModel<E> superTypeModel,
-            List<EntityModel<? extends E>> concreteModels
+    private PolymorphicEntityMessagingMetamodel(
+            EntityMessagingMetamodel<E> superTypeMetamodel,
+            List<EntityMessagingMetamodel<? extends E>> concreteMetamodels
     ) {
-        this.superTypeModel = Objects.requireNonNull(superTypeModel, "The superTypeModel may not be null.");
-        Objects.requireNonNull(concreteModels, "The concreteModels may not be null.");
-        this.concreteModels = new HashMap<>();
-        this.supportedCommandNames.addAll(superTypeModel.supportedCommands());
-        this.supportedInstanceCommandNames.addAll(this.superTypeModel.supportedInstanceCommands());
-        this.supportedCreationalCommandNames.addAll(superTypeModel.supportedCreationalCommands());
-        for (EntityModel<? extends E> polymorphicModel : concreteModels) {
-            this.concreteModels.put(polymorphicModel.entityType(), polymorphicModel);
+        this.superTypeMetamodel = Objects.requireNonNull(superTypeMetamodel, "The superTypeMetamodel may not be null.");
+        Objects.requireNonNull(concreteMetamodels, "The concreteMetamodels may not be null.");
+        this.concreteMetamodels = new HashMap<>();
+        this.supportedCommandNames.addAll(superTypeMetamodel.supportedCommands());
+        this.supportedInstanceCommandNames.addAll(this.superTypeMetamodel.supportedInstanceCommands());
+        this.supportedCreationalCommandNames.addAll(superTypeMetamodel.supportedCreationalCommands());
+        for (EntityMessagingMetamodel<? extends E> polymorphicModel : concreteMetamodels) {
+            this.concreteMetamodels.put(polymorphicModel.entityType(), polymorphicModel);
             this.supportedCommandNames.addAll(polymorphicModel.supportedCommands());
             this.supportedInstanceCommandNames.addAll(polymorphicModel.supportedInstanceCommands());
             this.supportedCreationalCommandNames.addAll(polymorphicModel.supportedCreationalCommands());
```

Comment:
I believe it should be this. Feel free to ignore if you left this on purpose:
```suggestion
        for (EntityMessagingMetamodel<? extends E> polymorphicMetamodel : concreteMetamodels) {
            this.concreteMetamodels.put(polymorphicMetamodel.entityType(), polymorphicMetamodel);
            this.supportedCommandNames.addAll(polymorphicMetamodel.supportedCommands());
            this.supportedInstanceCommandNames.addAll(polymorphicMetamodel.supportedInstanceCommands());
            this.supportedCreationalCommandNames.addAll(polymorphicMetamodel.supportedCreationalCommands());
```